api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -421,159 +424,191 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 	plannedCount := 0
 	planSuccessCount := 0
 	for _, step := range plan.Steps {
-		log.DebugfCtx(ctx, " M (Solution): processing step with Role %s on target %s", step.Role, step.Target)
-		for _, component := range step.Components {
-			log.DebugfCtx(ctx, " M (Solution): processing component %s with action %s", component.Component.Name, component.Action)
-		}
-		if s.IsTarget && !api_utils.ContainsString(s.TargetNames, step.Target) {
-			continue
-		}
-
-		if targetName != "" && targetName != step.Target {
-			continue
-		}
-
-		plannedCount++
-
-		dep.ActiveTarget = step.Target
-		agent := findAgentFromDeploymentState(mergedState, step.Target)
-		if agent != "" {
-			col[ENV_NAME] = agent
-		} else {
-			delete(col, ENV_NAME)
-		}
-		var override tgt.ITargetProvider
-		role := step.Role
-		if role == "container" {
-			role = "instance"
-		}
-		if v, ok := s.TargetProviders[role]; ok {
-			override = v
-		}
-		var provider providers.IProvider
-		if override == nil {
-			targetSpec := s.getTargetStateForStep(step, deployment, previousDesiredState)
-			provider, err = sp.CreateProviderForTargetRole(s.Context, step.Role, targetSpec, override)
+		select {
+		case <-ctx.Done():
+			// Context canceled or timed out
+			log.DebugfCtx(ctx, " M (Solution): reconcile canceled")
+			err = s.saveSummaryProgress(ctx, deployment.Instance.ObjectMeta.Name, deployment.Generation, deployment.Hash, summary, namespace)
```

Comment:
Why do we need to saveSummaryProgress here? I am afraid we need to persist SolutionManagerDeploymentState instead